### PR TITLE
[WIP] Mimic optimization procedure used in Flink

### PR DESCRIPTION
**Pull request by lmatz**
issue #1, moved some of the steps in Flink optimization procedure, still in progress. We may get the husky logical plan optimization almost done and then move to the last step --- husky physical plan optimization.

### CODE REVIEW

**Comment by TatianaJin** [comment]
How about breaking down the task and make small prs step by step? It would be easier to follow.

File: calcite/CMakeLists.txt [hunk 1]
```diff
@@ -27,3 +27,6 @@ install( FILES ${HuskySQL_JARS} DESTINATION share)
 # Executables
 add_custom_target(QueryPlan ALL mvn exec:java -Dexec.mainClass="husky.sql.calcite.planner.SimpleQueryPlanner" -Dexec.args=\"\'select units,color from Parts\'\" WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} DEPENDS ${HuskySQL_JARS})
 add_custom_target(ExecuteQuery ALL mvn exec:java -Dexec.mainClass="husky.sql.calcite.Execute" -Dexec.args=\"\'select units,color from Parts\'\" WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} DEPENDS ${HuskySQL_JARS})
+
+# Executables
+add_custom_target(HuskyQueryPlan ALL mvn exec:java -Dexec.mainClass="husky.sql.calcite.planner.HuskyQueryPlanner" -Dexec.args=\"\'select units,color from Parts\'\" WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} DEPENDS ${HuskySQL_JARS})
```
Comment (TatianaJin) [inline calcite/CMakeLists.txt:32]: may add this to travis test as well

**Comment by lmatz** [comment]
I will push smaller commits in the future. The current commit is sort of like the whole framework, has to push these so that the example can be runned. Most of part except the last step in current “getPhysicalPlan” func is standard and boilerplate. I may remove some parts which are not necessary at the current stage

File: calcite/CMakeLists.txt (hunk 1 above)
Comment (lmatz) [reply calcite/CMakeLists.txt:32]: done

**Pull request merged**